borgmatic/config/schema.yaml

```diff
@@ -1265,3 +1265,28 @@ properties:
                     Umask used when executing hooks. Defaults to the umask that
                     borgmatic is run with.
                 example: 0077
+            snapper:
+                type: object
+                required: ['include']
+                additionalProperties: false
+                properties:
+                    include:
```

Comment:
I think a good example would be that one has a btrfs filesystem and an ext4 filesystem and wants to backup both of them. Mainly I implemented it like this because it seemed way too inflexible to me to just have this as an all or nothing option.

Reply:
Makes sense.. Thanks for explaining. Do you have any thoughts on how this could/should interact with the built-in Borg/borgmatic excludes/patterns functionality? For instance, since all of this processing on `source_directories` happens before Borg is called, you could end up converting some excluded source directories into snapshot paths—which would then potentially no longer get excluded. Similarly, configured `patterns` could effectively include additional directories that can't use the snapper integration because they're not in `source_directories`.

Maybe these are super edge cases? Perhaps just known limitations to document?

Reply:
I think your `--dry-run` approach is a horrible idea performance-wise, but the alternative (handling parsing of the patterns, and matching that with available snapper configs) is a horrible idea code complexity-wise.
So I don't know what a good solution would be.

Reply:
Yeah, it's a thorny problem with this approach, and I'm not sure what I'd do there to solve it. The problem is that Borg basically has its own domain-specific language for what directories should be backed up. And if you'd like to _operate_ on that set of directories, you either need to get into the business of interpreting the domain-specific language or lean on Borg to do it for you.